### ISSUE TITLE
Annotation not imported

### ISSUE THREAD

**Issue by MinjieHu** [mgmeyers/obsidian-zotero-integration#102]
Hi,

When I try to create a research note, only the metadata of the paper is fetched, but the annotation is not included. See the example below:
<img width="710" alt="image" src="https://user-images.githubusercontent.com/39601931/194372246-0bf17c58-3cbc-404d-953b-c12eac0b9b3c.png">
I check the paper in Zotero and do find annotations there. Since I use a linked pdf, I opened the pdf file with the default system pdf viewer(Preview) and found there is no annotation in the pdf file. I guess zotero stored the annotation out of the pdf file which may lead to the issue. Can you suggest any way to get around this?

Thank you again!

**Comment by mgmeyers** [comment]
@MinjieHu That last screenshot is the citation style for the `Cite` command. The one used for imports looks like this:

<img width="698" alt="Screen Shot 2022-10-06 at 12 00 46 PM" src="https://user-images.githubusercontent.com/2694747/194396716-208c61c7-c804-4c31-9773-3cdc126cd7b8.png">

What is that set to?

**Comment by MinjieHu** [comment]
You are right. This set is not matching. When I change to the same bibliography style, this error is cleared. 

However, the annotation is still not fetched😥.

**Comment by MinjieHu** [comment]
It is weird when I try with four other papers, they all worked. All the one I tried before is not working.

**Comment by mgmeyers** [comment]
Hmm, any errors (red text) in Obsidian's developer console when you try to import that paper? You can open it by pressing cmd/ctrl + shift + i

**Comment by MinjieHu** [comment]
No, not any.

**Comment by mbojan** [comment]
I sadly confirm that importing annotations seems to stop working: no annotations in the data explorer (and hence in notes generated with a template that should handle them). I also noticed that one I try to "Create research note" and use the pop-up to search Zotero the results include the parent item and the PDF separately. I don't think it happened before (I may be wrong though).

I'll be happy to poke in my Obsidian and Zotero to debug this further. Just let me know where. It is a *crucial* functionality for me.

**Comment by MinjieHu** [comment]
@volker-fr 
Here are the attached I had for this particular entry:
<img width="945" alt="image" src="https://user-images.githubusercontent.com/39601931/195108465-958d1c5b-a527-40e6-99a0-ab6dd15699cb.png">

**Comment by volker-fr** [comment]
@mbojan & @MinjieHu can you try to update to the latest plugin version of Better Bibtex? The latest release version can be found here: https://github.com/retorquere/zotero-better-bibtex/releases   Zotero offered me to update it, but the version of the update wasn't the latest, I had to manually install the xpi file.

This solved my issue thanks to the fix from mgmeyers. See also https://github.com/mgmeyers/obsidian-zotero-integration/issues/107

**Comment by mbojan** [comment]
Thanks @volker-fr! Updating Better Bibtex (published now so can be updated in Zotero not from GitHub) indeed fixes the issue for me.

**Comment by MinjieHu** [comment]
@volker-fr after updating, it imports a figure when I highlighted. But the not annotation is still not imported. Since my other paper doesn't have this issue, this may be due to some specificity of this pdf file?

**Comment by volker-fr** [comment]
@MinjieHu Hand written annotations are indeed not imported. It's not implemented, but will no longer crash the import of the other annotation kinds.

I suggest to open a separate ticket/feature request for this.

**Comment by eztherwong** [comment]
I have the same issue. The zotero integration plugin can import metadata of all papers, but can only import annotations of some papers. Zotero Data Explorer listed 0 annotation items for papers with annotations that weren't imported. I'm using the latest version of everything. I don't see any error in the Obsidian's developer console.

I'm no programmer, but I guess if annotations can't be imported while metadata can, likely that it has something to do with the PDF Utility that is supposed to extract data from PDFs. 

@MinjieHu, any chance that you have found a fix? Thanks in advance!

**Comment by mgmeyers** [comment]
I've updated the plugin so that the data explorer and imports use the same code. Try updating to the latest version and see if this is still an issue.

**Comment by weztherw** [comment]
Thanks for the latest version. I've just tested it, but it still doesn't import annotations that I had tried to import before. The annotations are from a paper that is the only attachment in a parent item, and this parent item doesn’t share its Citation Key with any other item. Would it help if I provide an example paper through DM?
